client/my-sites/themes/controller-logged-in.jsx

```diff
@@ -47,3 +48,13 @@ export function upload( context, next ) {
 	context.primary = <Upload />;
 	next();
 }
+
+export function selectSiteIfLoggedIn( context, next ) {
+	const state = context.store.getState();
+	const currentUser = getCurrentUser( state );
+
+	if ( currentUser ) {
+		return sites( context, next );
```

Comment:
You want to also call `siteSelection` handler before `sites`. It has no actual site selection to perform, but if the user has only single site, it will redirect from `/themes` to `/themes/that.site` right away, without showing the selection UI.

The handler also shouldn't call `next` in case it decides to render the site selector. It wants to render the site selector and terminate the handler chain.
```js
if ( ! isUserLoggedIn( state ) {
  next();
  return;
}

siteSelection( context,
  () => sites( context,
    () => makeLayout( context,
      () => clientRender( context )
    )
  )
);
```